assets/js/modules/analytics-4/datastore/accounts.js

```diff
@@ -234,55 +254,103 @@ const baseActions = {
 
 		return matchedAccount || null;
 	},
+
+	/**
+	 * Creates an action to transform and sort account summaries.
+	 *
+	 * This action is typically dispatched when account summaries need to be
+	 * transformed (e.g., extracting and populating relevant account and property
+	 * IDs) and then sorted in a case-insensitive manner by display name.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} The action object with the type `TRANSFORM_AND_SORT_ACCOUNT_SUMMARIES`.
+	 */
+	transformAndSortAccountSummaries() {
+		return {
+			type: TRANSFORM_AND_SORT_ACCOUNT_SUMMARIES,
+		};
+	},
 };
 
 const baseControls = {};
 
-const baseReducer = ( state, { type } ) => {
+/**
+ * Creates immer reducer.
+ */
+const baseReducer = createReducer( ( state, { type } ) => {
 	switch ( type ) {
-		case START_SELECTING_ACCOUNT: {
-			return {
-				...state,
-				finishedSelectingAccount: false,
-			};
-		}
+		case START_SELECTING_ACCOUNT:
+			state.finishedSelectingAccount = false;
+			break;
+
+		case FINISH_SELECTING_ACCOUNT:
+			state.finishedSelectingAccount = true;
+			break;
+
+		case RESET_ACCOUNT_SUMMARIES:
+			state.accountSummaries = undefined;
+			state.settings.accountID = undefined;
+			state.settings.propertyID = undefined;
+			state.settings.measurementID = undefined;
+			state.settings.webDataStreamID = undefined;
+			break;
+
+		case TRANSFORM_AND_SORT_ACCOUNT_SUMMARIES:
+			if ( ! state.accountSummaries?.length ) {
+				return state;
+			}
 
-		case FINISH_SELECTING_ACCOUNT: {
-			return {
-				...state,
-				finishedSelectingAccount: true,
-			};
-		}
+			state.accountSummaries = state.accountSummaries.map(
+				( account ) => {
+					const accountObj = populateAccountID( account );
+					accountObj.propertySummaries = (
+						accountObj.propertySummaries || []
+					).map( ( property ) =>
+						populatePropertyAndAccountIds( property )
+					);
+
+					return accountObj;
+				}
+			);
 
-		case RESET_ACCOUNT_SUMMARIES: {
-			return {
-				...state,
-				accountSummaries: undefined,
-				settings: {
-					...state.settings,
-					accountID: undefined,
-					propertyID: undefined,
-					measurementID: undefined,
-					webDataStreamID: undefined,
-				},
-			};
-		}
+			state.accountSummaries = caseInsensitiveListSort(
+				state.accountSummaries,
+				'displayName'
+			);
 
-		default: {
 			return state;
-		}
 	}
-};
+} );
 
 const baseResolvers = {
 	*getAccountSummaries() {
 		const registry = yield commonActions.getRegistry();
+		let nextPageToken = null;
+		const errors = [];
 		const summaries = registry
 			.select( MODULES_ANALYTICS_4 )
 			.getAccountSummaries();
+
+		// Fetch initial account summaries if they are undefined.
 		if ( summaries === undefined ) {
-			yield fetchGetAccountSummariesStore.actions.fetchGetAccountSummaries();
+			do {
+				const { error, response } =
+					yield fetchGetAccountSummariesStore.actions.fetchGetAccountSummaries(
+						{
+							pageToken: nextPageToken,
+						}
```

Comment:
This doesn't work as expected since the fetch action expects a positional argument of just the token rather than this params-like object. It works for the initial request, but won't for subsequent pages.

See 
<img width="669" alt="image" src="https://github.com/user-attachments/assets/72848f7a-8b8c-4677-9195-31dc9cd46e60">

It should be as simple as this to fix (may need formatting)
```suggestion					
						nextPageToken
```

Reply:
@aaemnnosttv If we pass `nextPageToken` instead of `{pageToken: nextPageToken}`, the create-fetch-store throws the error 

```
TypeError: Cannot convert undefined or null to object
```

`fetchGetAccountSummaries` will always be called with `{pageToken: nextPageToken}` and we won't be passing string to it.